tests/test_explanation.py

```diff
@@ -58,3 +58,45 @@ def test_explanation_hstack_errors(random_seed):
             base_values=np.ones(20) * 0.987,
         )
         _ = base_exp.hstack(exp2)
+
+
+def test_square_explanation_feature_names_slicing_issue_2699(random_seed):
+    """Checks that feature names in Explanations are properly sliced with "square"
+    arrays (N==k).
+
+    For 2D arrays, there is an ambiguity in how to assign the feature names to the
+    slicer index. E.g. if feature_names is a list of 5 elements, and the shap_values is
+    a (5,5) array, it's ambiguous whether the axis=0 or axis=1 refers to the "feature
+    columns".
+
+    This test ensures that we give higher priority to axis=1 for the feature_names for
+    square arrays. Since most of the time, the 2D shap values arrays are assembled as
+    (# samples, # features).
+
+    cf. GH Issue #2722, #2699.
+    """
+    rs = np.random.RandomState(random_seed)
+    featnames = list("abcde")
+
+    exp = shap.Explanation(
+        # usually this arises as the shap values of N=6 samples, k=5 features
```

Comment:
Nit-nit : gramatically I wasn't sure what phenomenon is referened by _"usually this arises"_. We're talking about how values of that shape are interpreted, right?

Another optional idea: perhaps consider using `@pytest.mark.parametrize` to run this test for various shapes and input feature names, to see if there are any other odd edge cases?

Reply:
> We're talking about how values of that shape are interpreted, right?

yep, I was just trying to convey that a shap values array of shape `(6, 5)` typically arises from having 6 samples, 5 features. I've now modified the phrasing here slightly, hopefully it's a little better.

> perhaps consider using @pytest.mark.parametrize to run this test for various shapes and input feature names

Any particular examples in mind that you would like to test? In the context of the original issue, this test seems sufficient to me, but I suppose we could make the test more general.

Reply:
>Any particular examples in mind

I was just observing that the current test seems to be two similar halves, which could maybe be combined in a parameterised test. Something like:

```python
@parameterize(n_samples=[4,5,6])
def test_feature_names_slicing(n_samples):
    featnames = list("abcde")
    exp = shap.Explanation(
        values=np.ones((n_samples, 5)),
        feature_names=featnames,
        output_names=featnames,
    )
    first_sample = exp[0]
    assert first_sample.feature_names == first_sample.output_names == featnames
    column_e = exp[..., "e"]
    assert column_e.feature_names == "e"
```

I'm happy with either style, so will leave it to you to decide which you prefer.

Reply:
Thanks @connortann (sorry for the late reply, finally got around to finishing up this PR), I decided to go with your suggestion in the end.